### PR TITLE
Enable request tests on all platforms

### PR DESCRIPTION
**Pull request by kuhnroyal**
### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I'm adding
- [x] I have updated the documentation (if necessary)
- [x] I have run the tests without failures
- [x] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

File: dio/lib/src/adapters/browser_adapter.dart [hunk 1]
```diff
@@ -71,7 +71,9 @@ class BrowserHttpClientAdapter implements HttpClientAdapter {
           xhr.status!,
           headers: xhr.responseHeaders.map((k, v) => MapEntry(k, v.split(','))),
           statusMessage: xhr.statusText,
-          isRedirect: xhr.status == 302 || xhr.status == 301,
+          isRedirect: xhr.status == 302 ||
+              xhr.status == 301 ||
+              options.uri.toString() != xhr.responseUrl,
```
Comment (kuhnroyal) [inline dio/lib/src/adapters/browser_adapter.dart:76]: Judging from my tests, the XHR will never return 301/302 here as the browser or JS-engine does always follow redirects.

File: dio/test/request_integration_test.dart [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+          queryParameters: {'id': '12', 'name': 'wendu'},
+        );
+        expect(response.statusCode, 200);
+        expect(response.isRedirect, isFalse);
+        expect(response.data['method'], 'GET');
+        expect(response.data['args'], {'id': '12', 'name': 'wendu'});
+      });
+
+      // TODO This is not supported on web, should we warn?
+      test('GET with content', () async {
```
Comment (kuhnroyal) [inline dio/test/request_integration_test.dart:40]: GET with a body is not supported in web. What should we do here? Exception, warning?

File: dio/test/request_integration_test.dart [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+            'x-multi-value-request-header': ['value1', 'value2'],
+          },
+        ),
+      );
+      expect(response.statusCode, 200);
+      expect(response.isRedirect, isFalse);
+      expect(
+        response.data['headers']['X-Multi-Value-Request-Header'],
+        // TODO we have a diff here between browser and non-browser
+        equals('value1, value2'),
```
Comment (kuhnroyal) [inline dio/test/request_integration_test.dart:302]: Here we have different results on io & web, need to check what is going on.

File: dio/test/request_integration_test.dart (hunk 2 above)
Comment (AlexV525) [reply dio/test/request_integration_test.dart:38]: We should do nothing. It's under a policy as I remember, and XHR does not ring any bells when the requesting is sending a payload.

Comment (kuhnroyal) [reply dio/test/request_integration_test.dart:38]: > It's under a policy as I remember

Not sure what you mean by that.

> XHR does not ring any bells when the requesting is sending a payload

That is correct, the body data is just ignored. The dio API allows this but there is no feedback for the developer why it is not working. My concern is that people open bug reports for this :)

Comment (AlexV525) [reply dio/test/request_integration_test.dart:38]: > My concern is that people open bug reports for this :)

Adding warning logs like the `ImplyContentTypeInterceptor` might work then.

Comment (kuhnroyal) [reply dio/test/request_integration_test.dart:38]: I added a warning

File: dio/test/request_integration_test.dart (hunk 3 above)
Comment (kuhnroyal) [reply dio/test/request_integration_test.dart:300]: I adjusted the browser behaviour here, we were actually sending `List.toString()` which resulted in `[foo, bar]` formatting. Running the web request tests now also against `httpbun` detected this.

**Review by AlexV525 — APPROVED** [review]

**Pull request merged**